ui/app/js/controllers/user_edit_controller.js

```diff
@@ -148,7 +152,7 @@ function ($scope, $modalInstance, CSRF, Permissions, user, users) {
       })
       .error(function(response) {
         if (typeof response === 'object') {
-          $scope.errors = response;
+          $scope.errors.permission = response;
```

Comment:
I think that the error shown is reasonable as the "options" field will be incorrect. And to be able to fix that we can consider errors.permission in my opinion. Since this way we can take care of both incorrect "options" field and an unknown permission. It will also fix the case where a user re-enters the same permission for a given username.